### ISSUE TITLE
AddressSanitizer and/or valgrind runs

### ISSUE THREAD

**Issue by def-** [MaterializeInc/materialize#19934]
### Feature request

There have been issues with strange memory corruption, so something unsafe might be overwriting it. ASan and/or valgrind would be helpful. I'll investigate.

**Comment by uce** [comment]
@teskje @antiguru What's the priority of this for compute? We're planning to park this for now given that miri covers _some_ of what we wanted to get out of valgrind.

**Comment by teskje** [comment]
We mainly want to as much as possible lower the risk of getting hard-to-debug memory corruption issues in the future. It seems worth investing some time now to get continuous checking set up, if that helps us reduce the time (and stress) of debugging problems we could catch that way. I'm not sure how good of a replacement Miri is for ASan/Valgrind. If it can find most of the same issues and they are mostly redundant, we are probably fine with just Miri.

I'd expect that we need ASan/Valgrind at least to find bugs in the non-Rust dependencies we link in. One measure, if we don't want to spend time on setting up ASan/Valgrind, would be to introduce a policy that we only introduce new non-Rust dependencies when we are sufficiently sure they are stable, according to some tbd criteria. Having such a policy would probably be a good idea in any case.

Could Miri have caught the moka issue from last week?

**Comment by def-** [comment]
I don't think so, miri can't run the sufficiently complex cargo tests, and the simpler ones probably didn't expose the issue or use moka at all.
